src/data/cached-history.ts

```diff
@@ -158,8 +158,14 @@ export const getRecentWithCache = (
     }
     const stateHistory = computeHistory(hass, fetchedHistory, localize);
     if (appendingToCache) {
-      mergeLine(stateHistory.line, cache.data.line);
-      mergeTimeline(stateHistory.timeline, cache.data.timeline);
+      if (stateHistory.line.length) {
+        mergeLine(stateHistory.line, cache.data.line);
+      }
+      if (stateHistory.timeline.length) {
+        mergeTimeline(stateHistory.timeline, cache.data.timeline);
+        // Replace the timeline array to force an update
+        cache.data.timeline = [...cache.data.timeline];
```

Comment:
Since mutating the array doesn't trigger the update I did this to force it.  Is there a better way?

Reply:
I would expect `mergeLine` and `mergeTimeline` to create new arrays.

Reply:
`cache.data.timeline = mergeTimeline(stateHistory.timeline, cache.data.timeline);`

Reply:
That didn't work out very well unfortunately since the update is a partial update and the current code enumerates the new data so if the entity didn't update in history call it disappears.

Reply:
I ended up reverting the last commit because I think its too risky to refactor without adding a whole lot of tests.